plugins/openai_plugin.py

```diff
@@ -104,7 +108,11 @@ def request_http(self, query: dict, user_id: int, test_end_time: float = 0):
 
         result.start_time = time.time()
 
-        headers = {"Content-Type": "application/json"}
+        if self.enable_auth: 
+            headers = {"Content-Type": "application/json",
+                        "Authorization": f"Bearer {self.auth_token}"}
+        else: 
+            headers = {"Content-Type": "application/json"}
```

Comment:
```suggestion
        headers = {"Content-Type": "application/json"}
        
        if self.authorization: 
            headers["Authorization"] = f"Bearer {self.authorization}"
```
And likewise for next file